### PR TITLE
mangal: init at version 3.10.0

### PR DESCRIPTION
**Pull request by bertof**
Mangal is a fancy CLI app written in Go which scrapes, downloads and packs manga into different formats.

This pull request adds a derivation for Mangal

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/mangal/default.nix
Comment (smancill) [inline pkgs/applications/misc/mangal/default.nix:20]:
```suggestion
  postInstall = lib.optionalString (stdenv.hostPlatform == stdenv.buildPlatform) ''
```

File: pkgs/applications/misc/mangal/default.nix
Comment (smancill) [inline pkgs/applications/misc/mangal/default.nix:1]:
```suggestion
{ lib, stdenv, buildGoModule, fetchFromGitHub, installShellFiles }:
```

File: pkgs/applications/misc/mangal/default.nix
Comment (smancill) [inline pkgs/applications/misc/mangal/default.nix:34]: Missing `platforms`.

**Review by smancill — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/mangal/default.nix
Comment (smancill) [reply pkgs/applications/misc/mangal/default.nix:20]: This is required, as explained in the linked comment.

Comment (smancill) [reply pkgs/applications/misc/mangal/default.nix:20]: Well, not really hard-required, but why break cross-compilation when it can be supported from the start.

And, yes, the cross-compiled package will have a different output, but it cannot really be avoided, if to generate the completion you need to run the compiled `$out/bin/mangal` binary. I am kinda confused why you say it works.

Comment (bertof) [reply pkgs/applications/misc/mangal/default.nix:20]: Basically I don't know if the cross-compiling build environment also includes by default qemu to run the target application.
If it doesn't then your change is necessary to not break cross compilation. If it is, then it would just work.
In my case qemu is enabled in NixOS using `boot.binfmt.emulatedSystems = [ "armv7l-linux" "aarch64-linux" ];` in my config and the binary can be executed during the build and even after.

Comment (smancill) [reply pkgs/applications/misc/mangal/default.nix:20]: I trust the commenter in the link above requesting the fix for cross-compilation in another PR, who is a code owner of the Go compiler.

Comment (bertof) [reply pkgs/applications/misc/mangal/default.nix]: Ok, I updated the derivation as suggested. I'm still not that expert in Nix, that's why I'm often asking for more details

**Comment by smancill** [comment]
Result of `nixpkgs-review pr 178002` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mangal</li>
  </ul>
</details>

**Review by smancill — APPROVED** [review]

File: pkgs/applications/misc/mangal/default.nix
Comment (smancill) [reply pkgs/applications/misc/mangal/default.nix:38]: Sorry, I messed up, now I checked that if not set it will be set by default by `buildGoModule`, so you can remove it unless there is a reason to specify custom platforms.

Comment (bertof) [reply pkgs/applications/misc/mangal/default.nix:37]: No problem. I updated the derivation removing the entry.

**Comment by bertof** [comment]
I updated the derivation from version 3.3.0 to 3.4.1

**Comment by bertof** [comment]
I updated the derivation from version 3.4.1 to 3.5.0

**Comment by bertof** [comment]
@kalbasit any change to merge this pull request? It has already been reviewed and tested.

**Comment by bertof** [comment]
I updated the derivation from 3.5.0 to 3.6.0.
@smancill @kalbasit, if the pull request is ok could you please accept it?

**Comment by bertof** [comment]
I updated the derivation from 3.6.0 to 3.7.0.

**Comment by bertof** [comment]
Update derivation to 3.9.1

**Review by teto — APPROVED** [review]

File: pkgs/applications/misc/mangal/default.nix
Comment (marsam) [inline pkgs/applications/misc/mangal/default.nix:3]: There is no need to override go_1_18, since is the default version used in nixpkgs
```suggestion
{ lib, stdenv, buildGoModule, fetchFromGitHub, installShellFiles }:

buildGoModule rec {
```
besides that, LGTM

Comment (bertof) [reply pkgs/applications/misc/mangal/default.nix:3]: The default go version is 1.17 and some of the libraries use features that are only supported by 1.18+ . That's why I'm using the override

Comment (bertof) [reply pkgs/applications/misc/mangal/default.nix:3]: Never mind, my branch was too many commits behind and was still using 1.17 as default. Your suggestion is correct.

**Comment by marsam** [comment]
I'm getting a different hash, and seems it also happens in ofborg 
```
error: hash mismatch in fixed-output derivation '/nix/store/gp9qgm8vr8k9bbpn09y50hg4yy0zc57s-mangal-3.10.0-go-modules.drv':
         specified: sha256-s8pb0CEzQseNY6vyxS3pxeagX/xvRH9SHiCOOJIPkrk=
            got:    sha256-ZgeG8S78/yS9KfEK1bZ3VlQGmZPBo9TXLQLz8rXOEGo=
error: 1 dependencies of derivation '/nix/store/vqvrh5lkln9iz2856dxxp6vid3k4r45w-mangal-3.10.0.drv' failed to build
```
Would you mind updating the hash? I'll merge it after that

**Comment by bertof** [comment]
Can confirm. I guess the build process kept some old derivation as cache. Fixed now.

**Pull request merged**
